integration/aggregate_documents_compat_test.go

```diff
@@ -158,18 +158,17 @@ func testAggregateStagesCompatWithProviders(tt *testing.T, providers shareddata.
 				})
 			}
 
-			var t testtb.TB = tt
-			if tc.failsForSQLite != "" {
-				t = setup.FailsForSQLite(tt, tc.failsForSQLite)
+			if tc.failsForSQLite != "" && setup.IsSQLite(tt) {
```

Comment:
Other tests checked `setup.FailsForSQLite(tt, tc.failsForSQLite)`, but this test helper skips tests marked to fail for sqlite. Why is this handled differently? :thinking: Is it related to what you said about some collection failing and others collection not failing? Or because aggregation is not supported yet for sqlite? I'm assuming all aggregation tests are currently failing though.

A consequence is that when we start implementing aggregation for sqlite, we won't know which tests start passing, because they are skipped.

Reply:
This is done to handle failure correctly. The `failsForSQLite` function is used above in this function. Here, we need to return if we did the previous check, because right after this line we do a check for empty/non-empty results, which might be different if we expect test to fail.

Reply:
I see we check `failsForSQLite` earlier. And all sqlite would have emptyResult.